"invalid Yaml document separator" error with helm 3.7.1

When trying to install codecentric/keycloak:15.1.0 with helm 3.7.1 with 2 or more secrets, I've got this error:
> [ERROR] templates/secrets.yaml: unable to parse YAML: invalid Yaml document separator: apiVersion: v1

This error isn't triggered with helm 3.6.3

Corresponding `values.yaml` to reproduce:
```
secrets:
  db:
      data:
          DB_USER: YWRtaW4=
          DB_PASSWORD: cGFzc3dvcmQ=
  http:
      data:
          KEYCLOAK_USER: YWRtaW4=
          KEYCLOAK_PASSWORD: cGFzc3dvcmQ=
```

> $ helm lint   
> ==> Linting .
> [ERROR] templates/secrets.yaml: unable to parse YAML: invalid Yaml document separator: apiVersion: v1
> 
> Error: 1 chart(s) linted, 1 chart(s) failed

--- a/charts/keycloak/Chart.yaml
+++ b/charts/keycloak/Chart.yaml
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: keycloak
-version: 16.0.2
+version: 16.0.3
 appVersion: 15.0.2
 description: Open Source Identity and Access Management For Modern Applications and Services
 keywords:

--- a/charts/keycloak/templates/secrets.yaml
+++ b/charts/keycloak/templates/secrets.yaml
@@ -1,4 +1,5 @@
 {{- range $nameSuffix, $values := .Values.secrets -}}
+---
 apiVersion: v1
 kind: Secret
 metadata:
@@ -25,5 +26,4 @@ stringData:
   {{- printf "%s: %s" $key (tpl $value $ | quote) | nindent 2 }}
   {{- end }}
 {{- end }}
----
 {{- end -}}

